pyop2/petsc_base.py

```diff
@@ -601,6 +601,7 @@ class Mat(base.Mat):
     for each element in the :class:`Sparsity`."""
 
     def __init__(self, *args, **kwargs):
+        self.mat_type = kwargs.pop("mat_type", None)
```

Comment:
I found it strange that the OP2 Mat class didn't know the `mat_type` before.

In the future, it could be useful to inspect this for e.g. the aijmkl class from MKL.

Reply:
I guess ideally we would `setfromoptions` the petsc Mat under the hood. The issue is on the pyop2 side we need to do different things depending on the class of matrix. Hence all this special-casing...